Dockerfile

```diff
@@ -28,14 +28,27 @@ FROM golang:1.7.6
 #ENV https_proxy ""
 ARG http_proxy
 ARG https_proxy
-ENV GOPATH=/go/ NET_CONTAINER_BUILD=1
+
+WORKDIR /go/src/github.com/contiv/netplugin/
 
 ENTRYPOINT ["netplugin"]
 CMD ["--help"]
 
-COPY ./ /go/src/github.com/contiv/netplugin/
+# by far, most of the compilation time is building vendor packages
+# build the vendor dependencies as a separate docker caching layer
+COPY ./vendor/ /go/src/github.com/contiv/netplugin/vendor/
 
-WORKDIR /go/src/github.com/contiv/netplugin/
+RUN GOGC=1500 go install -ldflags "-s -w" $(go list ./vendor/...)
+
+# build the netplugin binaries
+COPY ./ /go/src/github.com/contiv/netplugin/
 
-RUN make build
+ARG BUILD_VERSION=""
+ARG NIGHTLY_RELEASE=""
 
+RUN GOPATH=/go/ \
```

Comment:
This Dockerfile should call `make deps` at some point to add the tools used to check if the source is fine to the image.

The executed command should include the checks target to run it before the build. This should reject code which doesn't meet the requirements.